### PR TITLE
Smoke test

### PR DESCRIPTION
**Pull request by eed3si9n**
This adds real-world integration test using scala/scala and Dotty compiler code base.
The expected parse success% is currently hardcoded, but at least we'll catch if this number degrades.

### CODE REVIEW

**Comment by eed3si9n** [comment]
Looks like this:
Smoke_test

File: script/smoke_test.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -e
+
+# This is an integration test to generally check the quality of parsing.
+
+SCALA_SCALA_LIBRARY_EXPECTED=87
+SCALA_SCALA_COMPILER_EXPECTED=46
+DOTTY_COMPILER_EXPECTED=35
```
Comment (keynmol) [inline script/smoke_test.sh:7]: _sad trombone music_

Comment (eed3si9n) [reply script/smoke_test.sh:7]: Only way from here is up. *optimistic trombone music*

Comment (SethTisue) [reply script/smoke_test.sh:7]: https://orchestraexcerpts.com/trombone-wagner-die-walkure-ride-valkyries-2-excerpts/

File: script/smoke_test.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+run_tree_sitter () {
+  local source_dir=$1
+  local expected=$2
+  echo "Parse $source_dir:"
+  out=$(tree-sitter parse "$source_dir"/**/*.scala --quiet --stat || true)
+  actual=$(echo $out | rev | cut -d' ' -f1 | rev | sed 's/.$//')
+  echo "$out"
+  if (( $(echo "$actual > $expected" |bc -l) )); then
+    echo -e "  - ${Green}${actual}% ok${Color_Off}"
```
Comment (keynmol) [inline script/smoke_test.sh:32]: Absolutely unnecessary, but would be nice to have: we could emit this green/red warning as a Github PR annotation: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-creating-an-annotation-for-an-error

Basically doing

```bash
echo "::error file=grammar.js,line=1::Missing semicolon"
```

Will show a message on the PR (similar to how coverage tools do it).

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -26,6 +45,13 @@ jobs:
           npm install && npm run build
           git diff --quiet || (echo "::error file=grammar.js::Generated parser differs from the checked in version" && exit 1)
 
-      - name: Parser tests 
+      - name: Parser tests
         shell: bash
         run: npm install && npm run test
+
+      - name: Smoke test
+        if: ${{ matrix.os == 'ubuntu-latest' }}
```
Comment (keynmol) [inline .github/workflows/ci.yml:53]:
```suggestion
        if: ${{ matrix.os == 'ubuntu-20.04' }}
```

Now that I think about it's probably easier to use `runner.os == "Linux"` if that's the right value https://docs.github.com/en/actions/learn-github-actions/environment-variables#detecting-the-operating-system

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -26,6 +45,13 @@ jobs:
...
+        shell: bash
+        run: |
+          export PATH="$PATH:node_modules/.bin/"
```
Comment (keynmol) [inline .github/workflows/ci.yml:56]: I believe you can avoid this and use `npm exec`, i.e.

```bash
npm exec -c 'tree-sitter --help'
```

Comment (eed3si9n) [reply .github/workflows/ci.yml:56]: Yea. Let me try that.

File: script/smoke_test.sh (hunk 2 above)
Comment (eed3si9n) [reply script/smoke_test.sh]: Done

Comment (keynmol) [reply script/smoke_test.sh]: Nice, I can see the correct messages in the actions log.

Your #82 will be the first one to demonstrate this working, as this PR doesn't touch grammar.js file so I guess the message just gets ignored.

**Comment by eed3si9n** [comment]
I've addressed @keynmol's reviews, so I am going to self-merge.

**Pull request merged**